### PR TITLE
Add TaskType Controller Infrastracture

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

**Comment by VLiamin** [comment]
It is better to remove "feature/" from the name of the PR. And make spaces for readability.

File: src/DashboardService.Business/TaskType/CreateTaskTypeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType;
+
+public class CreateTaskTypeCommand : ICreateTaskTypeCommand
+{
+  private readonly ITaskTypeRepository _repository;
+
+  public CreateTaskTypeCommand([FromServices] ITaskTypeRepository repository)
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/CreateTaskTypeCommand.cs]: You can remove [FromServices]  here

File: src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType;
+
+public class EditTaskTypeCommand : IEditTaskTypeCommand
+{
+  private readonly ITaskTypeRepository _repository;
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs:14]: empty line

File: src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  public EditTaskTypeCommand([FromServices] ITaskTypeRepository repository) {
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs]: '{' on next line

File: src/DashboardService.Validation/TaskType/CreateTaskTypeRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType;
+using LT.DigitalOffice.DashboardService.Validation.TaskType.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.DashboardService.Validation.TaskType;
+
+public class CreateTaskTypeRequestValidator : AbstractValidator<CreateTaskTypeRequest>, ICreateTaskTypeRequestValidator
+{
+
```
Comment (VLiamin) [inline src/DashboardService.Validation/TaskType/CreateTaskTypeRequestValidator.cs]: emty line

File: src/DashboardService.Validation/TaskType/Interfaces/ICreateTaskTypeRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+using FluentValidation;
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType;
+
+namespace LT.DigitalOffice.DashboardService.Validation.TaskType.Interfaces;
+
```
Comment (VLiamin) [inline src/DashboardService.Validation/TaskType/Interfaces/ICreateTaskTypeRequestValidator.cs:5]: empty line

File: src/DashboardService/Controllers/TaskTypeController.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    [FromBody] CreateTaskTypeRequest request,
+    [FromServices] CreateTaskTypeCommand command)
+  {
+    return await command.ExecuteAsync(request);
+  }
+
+  [HttpGet("get")]
+  public async Task<FindResultResponse<IEnumerable<TaskTypeInfo>>> GetAsync(
+    [FromServices] GetAllTaskTypesCommand command
+  )
```
Comment (VLiamin) [inline src/DashboardService/Controllers/TaskTypeController.cs]: on previous line

File: src/DashboardService.Business/TaskType/CreateTaskTypeCommand.cs (hunk 1 above)
Comment (Glunky) [reply src/DashboardService.Business/TaskType/CreateTaskTypeCommand.cs]: ++

File: src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs (hunk 2 above)
Comment (Glunky) [reply src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs:14]: ++

File: src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs (hunk 3 above)
Comment (Glunky) [reply src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs]: ++

File: src/DashboardService.Validation/TaskType/CreateTaskTypeRequestValidator.cs (hunk 4 above)
Comment (Glunky) [reply src/DashboardService.Validation/TaskType/CreateTaskTypeRequestValidator.cs]: ++

File: src/DashboardService.Validation/TaskType/Interfaces/ICreateTaskTypeRequestValidator.cs (hunk 5 above)
Comment (Glunky) [reply src/DashboardService.Validation/TaskType/Interfaces/ICreateTaskTypeRequestValidator.cs:5]: ++

File: src/DashboardService/Controllers/TaskTypeController.cs (hunk 6 above)
Comment (Glunky) [reply src/DashboardService/Controllers/TaskTypeController.cs]: ++

File: src/DashboardService.Business/TaskType/GetAllTaskTypesCommand.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType;
+
+public class GetAllTaskTypesCommand : IGetAllTaskTypesCommand
+{
+  private readonly ITaskTypeRepository _repository;
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/GetAllTaskTypesCommand.cs]: empty line

Comment (Glunky) [reply src/DashboardService.Business/TaskType/GetAllTaskTypesCommand.cs]: ++

File: src/DashboardService.Business/TaskType/GetTaskTypeCommand.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType;
+
+public class GetTaskTypeCommand : IGetTaskTypeCommand
+{
+  private readonly ITaskTypeRepository _repository;
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/GetTaskTypeCommand.cs:12]: line

**Review by VLiamin — APPROVED** [review]

File: src/DashboardService.Business/TaskType/GetTaskTypeCommand.cs (hunk 8 above)
Comment (Glunky) [reply src/DashboardService.Business/TaskType/GetTaskTypeCommand.cs:12]: ++

**Pull request closed without merging**